presto-main/src/main/java/com/facebook/presto/operator/aggregation/BindableAggregationFunction.java

```diff
@@ -124,9 +125,21 @@ public InternalAggregationFunction specialize(BoundVariables variables, int arit
 
         try {
             MethodHandle inputHandle = lookup().unreflect(inputFunction);
+            for (ScalarImplementation.ImplementationDependency dependency : implementation.get().getInputDependencies()) {
+                inputHandle = inputHandle.bindTo(dependency.resolve(variables, typeManager, functionRegistry));
+            }
+
             MethodHandle combineHandle = lookup().unreflect(combineFunction);
+            for (ScalarImplementation.ImplementationDependency dependency : implementation.get().getCombineDependencies()) {
+                combineHandle = combineHandle.bindTo(dependency.resolve(variables, typeManager, functionRegistry));
+            }
+
             // FIXME
             MethodHandle outputHandle = outputFunction == null ? null : lookup().unreflect(outputFunction);
+            for (ScalarImplementation.ImplementationDependency dependency : implementation.get().getOutputDependencies()) {
+                outputHandle = outputHandle.bindTo(dependency.resolve(variables, typeManager, functionRegistry));
+            }
```

Comment:
Honestly, I believe the way OperatorDependency is implemented for scalars today is broken. It doesn't work together with functions with instance factories. It's unfortunate we moved on to aggregation functions before we get a chance to fix that. Anyways, that issue is parallel to this pull request.